testng-core/src/main/java/org/testng/DependencyMap.java

```diff
@@ -64,26 +65,32 @@ public ITestNGMethod getMethodDependingOn(String methodName, ITestNGMethod fromM
     if (l.isEmpty() && fromMethod.ignoreMissingDependencies()) {
       return fromMethod;
     }
-    Optional<ITestNGMethod> found =
-        l.stream()
-            .parallel()
-            .filter(
-                m ->
-                    isSameInstance(fromMethod, m)
-                        || belongToDifferentClassHierarchy(fromMethod, m)
-                        || hasInstance(fromMethod, m))
-            .findFirst();
-    if (found.isPresent()) {
-      return found.get();
-    }
+    return l.parallelStream()
+        .filter(
+            m ->
+                isSameInstance(fromMethod, m)
+                    || belongToDifferentClassHierarchy(fromMethod, m)
+                    || hasInstance(fromMethod, m))
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new TestNGException(
+                    "Method \""
+                        + fromMethod
+                        + "\" depends on nonexistent method \""
+                        + methodName
+                        + "\""));
+  }
 
-    throw new TestNGException(
-        "Method \"" + fromMethod + "\" depends on nonexistent method \"" + methodName + "\"");
+  private static boolean methodNameMatch(Pattern pattern, String method) {
+    String justMethodName = method.substring(method.lastIndexOf(".") + 1);
```

Comment:
Instead of being less precise, why not build the pattern with the class name?

Could you try some samples where there is inner classes and methods with identical names?
I feel unexpected behavior in some cases.

Reply:
@juherr 

> Could you try some samples where there is inner classes

I didnt quite understand this part. Can you share a sample that shows what you are talking about ?

Reply:
@juherr ping! Can U pls let me know?

Reply:
Are we sure that a method with dependency will not found its regex dependency from an inner class?

Class with a test method that dependsOn x*
Inner class with xx test method.
The test from parent should fail because it don't find its dependency.